### PR TITLE
post-install: add post-install hooks support

### PR DESCRIPTION
**Pull request by dorileo**
Introduce the initial support to post-install hooks. The code also
expands a set of pre-defined variables, for now the only exposed
variable is the chrootDir (the directory where the target system is
mounted).

Signed-off-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>

### CODE REVIEW

**Comment by dorileo** [comment]
@mdhorn added the pre-install hooks

File: tests/valid-minimal.yaml [hunk 1]
```diff
@@ -23,3 +23,13 @@ telemetry: false
 keyboard: us
 language: en_US.UTF-8
 kernel: kernel-native
+pre-install: [
+   {cmd: 'echo "running pre-install hook form: $chrootDir"'}
```
Comment (mdhorn) [inline tests/valid-minimal.yaml]: is chrootDir set to "/" when chroot: false ?

Comment (dorileo) [reply tests/valid-minimal.yaml]: nope

**Comment by dorileo** [comment]
@mdhorn updated, thanks.

**Pull request merged**